Update EIP-721: Add ERC links

fulldecent opened the pull request:
This fixes some linter complaints

Here is a validation run that reporting the error which this PR fixes.

https://github.com/ethereum/EIPs/actions/runs/6045138001/job/16404727259?pr=6527

Pandapip1 reviewed (changes requested):
ERC-721 should be replaced with "This EIP" instead of self-referencing by number.

Library links aren't supported. Use relative links instead (i.e. `[ERC-20](./eip-20.md)` instead of `[ERC-20](eip-20)`.

fulldecent commented:
@Pandapip1 I have updated all references to "ERC-721" from within that document to "this ERC". Some of this required a little disambiguation because this document is referring to an ERC (numbered 721) an interface (named `ERC721`) and two other interfaces (`ERC721Enumerable`, `ERC721Metadata`). My action here was to maintain clarity about what was being discussed while minimizing word changes.

This document previously refers to these things as "ERC"s. I kept this nomenclature in the word changes above. If there will consideration of changing this to "EIP", please let's separate that to a different, independent discussion and PR.

A reference was previously made to 0xcert's implementation which is named ERC-721. I updated this to its new name "Nibbstack". But the new product is still named `ERC721`. Probably this will also trigger a false-positive lint.

Pandapip1 commented on EIPS/eip-721.md:
```diff
@@ -28,17 +28,17 @@ In general, all houses are distinct and no two kittens are alike. NFTs are *dist
+A standard interface allows wallet/broker/auction applications to work with any NFT on Ethereum. We provide for simple non-fungible token smart contracts as well as contracts that track an *arbitrarily large* number of NFTs. Additional applications are discussed below.
 
-This standard is inspired by the [ERC-20](eip-20) token standard and builds on two years of experience since EIP-20 was created. EIP-20 is insufficient for tracking NFTs because each asset is distinct (non-fungible) whereas each of a quantity of tokens is identical (fungible).
+This standard is inspired by the [ERC-20](./eip-20) token standard and builds on two years of experience since EIP-20 was created. EIP-20 is insufficient for tracking NFTs because each asset is distinct (non-fungible) whereas each of a quantity of tokens is identical (fungible).
```
```suggestion
This standard is inspired by the [ERC-20](./eip-20.md) token standard and builds on two years of experience since EIP-20 was created. EIP-20 is insufficient for tracking NFTs because each asset is distinct (non-fungible) whereas each of a quantity of tokens is identical (fungible).
```

Pandapip1 reviewed (dismissed):
+1 if my change fixed the errors

fulldecent commented:
@Pandapip1 thank you, yes that's good now!

SamWilsn reviewed (changes requested):
I undid a few content changes, and fixed a few more `eipw` errors. Still a bunch of them because of the external links.

I'll add this to tomorrow's EIPIP call agenda.

bumblefudge commented:
thanks for this

SamWilsn commented:
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.
